src/coreclr/interpreter/compiler.cpp

```diff
@@ -1022,6 +1069,95 @@ void InterpCompiler::BuildGCInfo(InterpMethod *pInterpMethod)
 #endif
 }
 
+void InterpCompiler::GetNativeRangeForClause(uint32_t startILOffset, uint32_t endILOffset, int32_t *nativeStartOffset, int32_t* nativeEndOffset)
+{
+    InterpBasicBlock* pStartBB = m_ppOffsetToBB[startILOffset];
+    InterpBasicBlock* pEndBB = pStartBB;
+    for (InterpBasicBlock* pBB = pStartBB->pNextBB; (pBB != NULL) && ((uint32_t)pBB->ilOffset < endILOffset); pBB = pBB->pNextBB)
+    {
+        if ((pBB->clauseType == pStartBB->clauseType) && (pBB->overlappingEHClauseCount == pStartBB->overlappingEHClauseCount))
+        {
+            pEndBB = pBB;
+        }
+    }
+
+    *nativeStartOffset = pStartBB->nativeOffset;
+    *nativeEndOffset = pEndBB->nativeEndOffset;
+}
+
+void InterpCompiler::BuildEHInfo()
+{
+    uint32_t lastTryILOffset = 0;
+    uint32_t lastTryILLength = 0;
+
+    INTERP_DUMP("EH info:\n");
+
+    if (m_methodInfo->EHcount == 0)
+    {
+        INTERP_DUMP("  None\n");
+        return;
+    }
+
+    m_compHnd->setEHcount(m_methodInfo->EHcount);
+    for (unsigned int i = 0; i < m_methodInfo->EHcount; i++)
+    {
+        CORINFO_EH_CLAUSE clause;
+        CORINFO_EH_CLAUSE nativeClause;
+
+        m_compHnd->getEHinfo(m_methodInfo->ftn, i, &clause);
+
+        int32_t tryStartNativeOffset;
+        int32_t tryEndNativeOffset;
+        GetNativeRangeForClause(clause.TryOffset, clause.TryOffset + clause.TryLength, &tryStartNativeOffset, &tryEndNativeOffset);
+
+        int32_t handlerStartNativeOffset;
+        int32_t handlerEndNativeOffset;
+        GetNativeRangeForClause(clause.HandlerOffset, clause.HandlerOffset + clause.HandlerLength, &handlerStartNativeOffset, &handlerEndNativeOffset);
+
+        nativeClause.TryOffset = ConvertOffset(tryStartNativeOffset);
+        nativeClause.TryLength = ConvertOffset(tryEndNativeOffset);
```

Comment:
these say length but are having end offset stored into them

Reply:
OK. In that case I see some places where we are doing Offset + Length, I will flag them

Reply:
Ah, no. The clauses coming from IL really have length, but the clauses that runtime expects (the output) have end offset in the length field.

Reply:
In the interpreter compiler, all the clause usages use the "Length" field to mean length. Only the nativeClause instance in this method is the "Length" field means end case.